src/main/java/blusunrize/immersiveengineering/client/models/smart/ConnModelReal.java

```diff
@@ -183,16 +183,22 @@ public boolean equals(Object obj)
 			if (!(obj instanceof ExtBlockstateAdapter))
 				return false;
 			ExtBlockstateAdapter o = (ExtBlockstateAdapter) obj;
-			for (IProperty i : state.getPropertyNames())
+			for (IProperty<?> i : state.getPropertyNames())
 			{
+				if (!o.state.getProperties().containsKey(i))
+					return false;
 				Object valOther = o.state.getValue(i);
-				if (valOther == null || !valOther.equals(state.getValue(i)))
+				Object valMine = o.state.getValue(i);
```

Comment:
shouldn't this just be "state." instead of "o.state"? Otherwise you're comparing the other to itself.